### ISSUE TITLE
macOS version

### ISSUE THREAD

**Issue by muveso** [unitycoder/UnityLauncherPro#109]
are you planning to release a macOS version of the launcher?

**Comment by Mythique** [comment]
I'm also interested about this, so I looked around a bit.

The project is currently on .NET Framework, which is not compatible with Mac OS. To make it work it would need to be [ported to .NET 6 or 7](https://learn.microsoft.com/en-us/dotnet/core/porting/upgrade-assistant-overview).

[The process to port a WPF app to .NET is described here](https://learn.microsoft.com/en-us/dotnet/core/porting/upgrade-assistant-wpf-framework). I ran the analysis tool, there's not a lot to fix before upgrading :

`[21:30:19 INF] Diagnostic UA0013_M with the message Windows Forms Deprecated controls : DataGrid needs to be replaced with DataGridView. For more details see - https://docs.microsoft.com/en-us/dotnet/core/compatibility/winforms#removed-controls generated`

`[21:30:19 WRN] Default font in Windows Forms has been changed from Microsoft Sans Serif to Seg Segoe UI, in order to change the default font use the API - Application.SetDefaultFont(Font font). For more details see here - https://devblogs.microsoft.com/dotnet/whats-new-in-windows-forms-in-net-6-0-preview-5/#application-wide-default-font.`

With that said I guess there is custom code that is specifically designed to work on Windows and would need to be refactored to work on Mac OS.

And I personally don't know if there are any downside to port a project from .NET Framework to .NET.

**Comment by unitycoder** [comment]
definitely interested, but haven't found suitable frameworks.. havent seen that ^ one before, i'll check it.

here's some previous notes:
https://github.com/unitycoder/UnityLauncherPro/issues/25

for mac, the native launcher is best apparently: https://github.com/unitycoder/UnityLauncherPro/wiki/Alternative-Launchers

**Comment by muveso** [comment]
I believe macOS support is possible with little changes, if you have time to do that, it would be very good

**Comment by unitycoder** [comment]
Tested running that wpf upgrader (on cloned foldeR), at least application still runs. (after disabled some parts that need upgrading).

how to make macos builds from this .net7 project though? or need to build in mac itself i guess?

**Comment by muveso** [comment]
we can help you to build mac app I guess

**Comment by Mythique** [comment]
Unfortunately I think that WPF is strictly for Windows, and we would need to use another [UI framework such as Avalionia](https://github.com/AvaloniaUI/Avalonia).

**Comment by muveso** [comment]
I believe [Avalonia](https://github.com/AvaloniaUI/Avalonia) works better. Actually, Unity Hub sucks. **it takes almost 20-30 seconds to open up** and it is honestly not functional.

If I didn't see **UnityLauncherPro**, I would develop such a launcher tool. But **UnityLauncherPro** is a handy and well-planned tool; if you have time to upgrade it to use both platforms, it will be tremendous.

If you don't have much time, I will develop from starch.

**Comment by unitycoder** [comment]
Yeah, unlikely that i'll do full conversion at the moment,
Avalonia looks interesting (and it says unity plastic client was done with it too), going to test it briefly still.
MAUI ive tried but it was quite limited and slow itself..

**Comment by muveso** [comment]
okay if you have a branch to work in Avalonia, I can support you anytime you want.

**Comment by unitycoder** [comment]
@muveso added you here https://github.com/unitycoder/UnityLauncherAvalonia so can play freely with the project there

**Comment by muveso** [comment]
okay thank you, I will manage it

**Comment by unitycoder** [comment]
something new here

> We enable WPF apps to run on macOS and Linux with our Fork of WPF, which allows us to maintain API compatibility.

https://avaloniaui.net/XPF

**Comment by unitycoder** [comment]
new one (but requires js+rust programming probably..)
https://tauri.app/

**Comment by hifron** [comment]
what about making a Crowdfunding Request on some platform(like kickstarter.com) or even make some suggestion and be somehow related with Unity corporation to get(but also set) some request how could Unity be?

And if Unity is multi-platform, launcher could be also but also from command-line and that is area where Unity suffers and shouldn't

**Comment by unitycoder** [comment]
yup, unity uses Electron for their Hub, so its easier to make multiplatform.. but its slow & bloated then.

UnityHubNative would be the way to go, it should be fast everywhere.. (uses c++ which i don't really know about)
https://github.com/unitycoder/UnityLauncherPro/wiki/Alternative-Launchers

i've tested those previously mentioned frameworks in earlier messages, all of them are worse than WPF..
so still waiting for better solutions.

Tauri was bit promising, but cannot make simple single exe builds with it..
 https://github.com/unitycoder/tauri_ulaunch/issues/2#issuecomment-1976413051
test project: https://github.com/unitycoder/tauri_ulaunch

**Comment by Mythique** [comment]
> something new here
> 
> > We enable WPF apps to run on macOS and Linux with our Fork of WPF, which allows us to maintain API compatibility.
> 
> https://avaloniaui.net/XPF

I'm curious, what was the issue with this solution?

**Comment by unitycoder** [comment]
does imgui has all the necessary components (alternatives to: datagrids, dropdowns, other regular UI elements) ?

the avalonia XPF, never received trial access yet.. so that one is still potential.

**Comment by Mythique** [comment]
You could try the [dear imgui demo](https://github.com/ocornut/imgui?tab=readme-ov-file#demo). I've seen dropdowns, datagrids and more ui stuff. 

It's code oriented though, so you wouldn't have a visual editor (unless someone made a tool for this). 

I wanted to ask for the xpf but I saw the price... Not sure if it could be free for an open source project? If price isn't an issue, it could be better than rewriting the ui indeed.

**Comment by unitycoder** [comment]
bumped into another framework,
https://platform.uno/

https://platform.uno/docs/articles/wpf-migration.html
but i don't understand why "migrating WPF apps to web".. i don't want web, just multiplatform desktop..

**Comment by Mythique** [comment]
[PanGui](https://pangui.io/) could also be something interesting. It's not released yet though.

**Comment by unitycoder** [comment]
Avalonia indie is now available... 1000usd Per app, per year subscription : O 
https://avaloniaui.net/xpf#pricing

**Comment by hifron** [comment]
WPF means windows forms aka Windows UI which is protected windows technology not available for other platforms like webview2, but html applications could be made if tech stack is compatible so html desktop multiplatform applications are now possible if decent programmers make them, but FluentUI(and WPF) is not.

**Comment by unitycoder** [comment]
yes, still looking for perfect platform..

Requirements or Wish List:
- Fast startup (unlike electron apps)
- Not browser based (takes so much memory to have full browser embedded and sluggish in general?)
- Single exe builds would be nice
- Standard UI components available (buttons, tabs, dropdown menu, datagrid or alternatives, checkboxes etc)
- Runs in pc, mac, linux
- Using C#

**Comment by feipinxiang** [comment]
> yes, still looking for perfect platform..是的，仍在寻找完美的平台。
> 
> Requirements or Wish List:要求或愿望清单：
> 
> * Fast startup (unlike electron apps)快速启动（与电子应用程序不同）
> * Not browser based (takes so much memory to have full browser embedded and sluggish in general?)不是基于浏览器的（嵌入完整的浏览器会占用太多内存，而且总体来说运行缓慢？）
> * Single exe builds would be nice单个 exe 版本会很好
> * Standard UI components available (buttons, tabs, dropdown menu, datagrid or alternatives, checkboxes etc)可用的标准 UI 组件（按钮、选项卡、下拉菜单、数据网格或替代方案、复选框等）
> * Runs in pc, mac, linux可在 PC、Mac、Linux 上运行
> * Using C#  使用 C#

Is there any progress on the mac version?

**Comment by unitycoder** [comment]
> Is there any progress on the mac version?

No news..

this seemed promising, but too expensive 
https://github.com/unitycoder/UnityLauncherPro/issues/109#issuecomment-2435795942


*I would estimate that AI can do good enough conversion into some multi-platform sdk within few years : ) /RemindMeIn2Years
